### PR TITLE
Refresh package list after pushing a package

### PR DESCRIPTION
**Pull request by nl0**

### CODE REVIEW

**Comment by nl0** [comment]
follow-up to #1843

File: catalog/app/containers/Bucket/PackageList.js [hunk 1]
```diff
@@ -386,7 +391,7 @@ export default function PackageList({
 
   return (
     <>
-      <UploadDialog bucket={bucket} open={uploadOpen} onClose={closeUpload} />
+      <UploadDialog {...{ bucket, refresh, open: uploadOpen, onClose: closeUpload }} />
```
Comment (fiskus) [inline catalog/app/containers/Bucket/PackageList.js:394]: I think that names were implied to be "**openUpload**" and "closeUpload"

File: catalog/app/containers/Bucket/UploadDialog.js [hunk 2]
```diff
@@ -219,11 +221,12 @@ function FilesInput({ input, meta, uploads, setUploads, errors = {} }) {
     setUploads(R.dissoc(path))
   }
 
+  const onInputChange = input.onChange
```
Comment (fiskus) [inline catalog/app/containers/Bucket/UploadDialog.js:224]: `input.onChange` used in onDrop callback too. Maybe you want to use `onInputChange` there too

File: catalog/app/containers/Bucket/UploadDialog.js [hunk 3]
```diff
@@ -758,6 +762,11 @@ export default function UploadDialog({ bucket, open, onClose }) {
           meta: getMetaValue(meta),
         },
       })
+      if (refresh) {
+        // wait for ES index to receive the new package data
+        await new Promise((resolve) => setTimeout(resolve, ES_LAG))
+        refresh()
```
Comment (fiskus) [inline catalog/app/containers/Bucket/UploadDialog.js:768]: Hm... I would expect, that `refresh` reloads page or reloads some component. If it reloads data, I would use `reFetch`, or `reloadData`, or `retryFetching`

**Review by fiskus — APPROVED** [review]
I didn't dig into `Data.use` yet

File: catalog/app/containers/Bucket/PackageList.js (hunk 1 above)
Comment (nl0) [reply catalog/app/containers/Bucket/PackageList.js:394]: `uploadOpen` is a state, more like `uploadDialogIsOpen`, and `closeUpload` is an action/verb

File: catalog/app/containers/Bucket/UploadDialog.js (hunk 3 above)
Comment (nl0) [reply catalog/app/containers/Bucket/UploadDialog.js:768]: well, it _refreshes_ the data, so i think it's ok semantically)

File: catalog/app/containers/Bucket/UploadDialog.js (hunk 2 above)
Comment (nl0) [reply catalog/app/containers/Bucket/UploadDialog.js:224]: k

File: catalog/app/containers/Bucket/UploadDialog.js (hunk 3 above)
Comment (fiskus) [reply catalog/app/containers/Bucket/UploadDialog.js:773]: Of course. Just sound my thoughts :)

**Review by akarve — APPROVED** [review]

**Pull request merged**